src/production/module/ImportReference.ts

```diff
@@ -0,0 +1,268 @@
+import { IVisit } from '../../compiler/Visitor/Visitor';
+import { ASTNode } from '../../compiler/interfaces/AST';
+import { Module } from '../../core/Module';
+import { IProductionContext } from '../ProductionContext';
+
+export type IImportReferences = ReturnType<typeof ImportReferences>;
+
+export enum ImportReferenceType {
+  SIDE_EFFECT_IMPORT,
+  IMPORT_SPECIFIERS,
+  DYNAMIC_IMPORT,
+  EXPORT_FROM,
+};
+
+export interface ImportReferencesProps {
+  module: Module;
+  productionContext: IProductionContext;
+  visit: IVisit;
+};
+
+export interface ImportReferenceProps {
+  module: Module;
+  source: string;
+  specifiers?: Array<ASTNode>;
+  type: ImportReferenceType;
+  visit: IVisit;
+};
+
+export function ImportReference(props: ImportReferenceProps) {
+  const exported = {
+    module: props.module,
+    remove: function () {
+      // temporary added for test purposes
+      exported.removed = true;
+
+      // @todo remove myself..
+      if (props.visit.property && props.visit.parent) {
+        if (props.visit.parent[props.visit.property] instanceof Array) {
+          const index = props.visit.parent[props.visit.property].indexOf(props.visit.node);
+          if (index > -1) {
+            props.visit.parent[props.visit.property].splice(index, 1);
+          }
+        }
+      }
+    },
+    // @todo remove this when we're implementing remove
+    removed: false,
+
+    source: props.source,
+    specifiers: props.specifiers,
+    type: props.type,
+    visit: props.visit
+  };
+
+  return exported;
+};
+
+export interface ImportSpecifierReferenceProps {
+  local: string;
+  name: string;
+  specifier: ASTNode;
+  visit: IVisit;
+};
+
+export function ImportSpecifierReference(props: ImportSpecifierReferenceProps) {
+  return {
+    local: props.local,
+    name: props.name,
+    remove: function () {
+      // @todo
+      if (props.visit.node.specifiers instanceof Array) {
+        const index = props.visit.node.specifiers.indexOf(props.specifier);
+        if (index > -1) {
+          props.visit.node.specifiers.splice(index, 1);
+        }
+      }
+    },
+    specifier: props.specifier,
+    visit: props.visit
+  }
+}
+
+
+function ImportSpecifier(visit: IVisit, specifier: ASTNode): ImportSpecifierReference {
+  let local: string;
+  let name: string;
+
+  if (specifier.type === 'ImportNamespaceSpecifier') {
+    // import * as React from 'react'
+    local = specifier.local.name;
+    name = 'default';
```

Comment:
oops, `name` shouldn't even be here. This will lead to errors, We should instead set `importNamespace : true` IMHO.
but name should be definitely `undefined`
Suggestions
`importNamespace` `importAll`

Reply:
added types for specifiers:

```ts
export enum ImportSpecifierReferenceType {
  OBJECT_SPECIFIER,
  NAMESPACE_SPECIFIER
}
```